### PR TITLE
Retrying port connections if failed

### PR DESCRIPTION
**Pull request by e-kolpakov**

### CODE REVIEW

File: files/sanity_check.py [hunk 1]
```diff
@@ -94,11 +111,14 @@ def test_global_free_percentage(default_free_space_percentage):
 
 def test_ports(ports):
     response = []
-    for port in ports:
-        if not check_port(port['host'], int(port['port'])):
+    for port_config in ports:
+        host, port = port_config['host'], int(port_config['port'])
+        retries = int(port_config.get('retries', DEFAULT_PORT_RETRIES))
+        retry_delay = int(port_config.get('retry_delay', DEFAULT_PORT_RETRY_DELAY))
```
Comment (bdero) [inline files/sanity_check.py]: Could you please add `retries` and `retry_delay` as part of the example configuration in the readme?

**Review by bdero — COMMENTED** [review]
I looked through the code and it looks good, I just have one comment. Also, how did you go about testing this/what is the easiest way for me to test this in your opinion? I'd like to test this more closely before I give this a +1.

**Comment by e-kolpakov** [comment]
@bdero The easiest would be to take this new version of sanity_check.py, copy it to some temp folder, replace/comment parts of `sanity_check` method that are irrelevant (non-port tests) or unavailable (reporting to snitch/email) give it a json file with config and run it.

Here's my config:

``` json
{
    "ports": [
        {
            "host": "localhost",
            "port": 8000,
            "message": "QWERTY",
            "retries": 1,
            "retry_delay": 5
        }
    ]
}
```

And I launched `python -m SimpleHTTPServer` in other console tab when checking "connection successful" case.

**Comment by bdero** [comment]
@e-kolpakov On the new dalite instance, I shutdown gunicorn and gave the script a spin, but both the syslog and my direct running attempts reveal that it always seems to be returning 0. Can you take a look? In the script's current state, it looks like we won't get emailed if gunicorn is down:

```
root@dalite:/home/ubuntu# service gunicorn stop
 * Stopping Gunicorn workers
 [dalite] *
root@dalite:/home/ubuntu# ps faux | grep gunicorn
root      7445  0.0  0.0  10460   936 pts/1    S+   15:13   0:00                          \_ grep --color=auto gunicorn
root@dalite:/home/ubuntu# /usr/local/sbin/sanity-check.py /root/sanity-check.json
root@dalite:/home/ubuntu# echo $?
0
```

**Comment by bdero** [comment]
@e-kolpakov I didn't notice that the script calls the [mail command](https://github.com/open-craft/ansible-sanity-checker/blob/2bcdc03333a3b8fe6a7f7f045ed21f60db485e0f/files/sanity_check.py#L79-L85). We received emails for it. I finished testing and it I couldn't find any regressions. :)

:+1:
- [x] I tested this: On the new dalite instance, I shutdown gunicorn to ensure that emails would be send out, and I confirmed that it takes more time to fail than the older script.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ N/A
- [x] Includes documentation

**Pull request merged**